include/cadmium/core/simulation/parallel_root_coordinator.hpp

```diff
@@ -38,177 +40,388 @@ namespace cadmium {
+                //get number of internal couplings
+                auto n_internal_couplings = internal_couplings.size();
+                double localNext;
+
+                while(timeNext < timeFinal) {
+                    // Step 1: execute output functions
+					#pragma omp for schedule(static)
+                    for(size_t i=0; i<n_subcomponents;i++){
+                    	subcomponents.at(i)->collection(timeNext);
                     }
-#pragma omp barrier
+					#pragma omp barrier
                     //end Step 1
 
-                    // Step 2: route messages
-#pragma omp for schedule(static)
+/*
+					// Step 2: route messages
+					#pragma omp for schedule(static)
+					for(size_t i=0; i<n_internal_couplings;i++) {
+                    	internal_couplings.at(i).first->parallelPropagate(internal_couplings.at(i).second);
+                    }
+                    // end Step 2
+*/
+            		// Step 2: route messages
+					#pragma omp for schedule(static)
                     for(size_t i=0; i<n_internal_couplings;i++){
-                        std::get<1>(internal_couplings.at(i))->parallelPropagate(std::get<0>(internal_couplings.at(i)));
+                    	for(auto& portFrom: internal_couplings[i].second){
+                    		internal_couplings.at(i).first->propagate(portFrom);
+                    	}
                     }
-#pragma omp barrier
+					#pragma omp barrier
                     // end Step 2
 
                     // Step 3: state transitions
-#pragma omp for schedule(static)
+					#pragma omp for schedule(static)
                     for(size_t i=0; i<n_subcomponents;i++){
                         subcomponents.at(i)->transition(timeNext);
                         subcomponents.at(i)->clear();
                     }
-#pragma omp barrier
+					#pragma omp barrier
                     // end Step 3
 
                     // Step 4: time for next events
                     localNext = subcomponents[0]->getTimeNext();
-#pragma omp for schedule(static)
+					#pragma omp for schedule(static)
                     for(size_t i=1; i<n_subcomponents;i++){
                         if(subcomponents[i]->getTimeNext() < localNext){
                             localNext = subcomponents[i]->getTimeNext();
                         }
                     }
-#pragma omp single
+					#pragma omp single
                     {
                         timeNext = localNext;
                     }
-#pragma omp barrier
-#pragma omp critical
+					#pragma omp barrier
+					#pragma omp critical
                     {
                         if(localNext < timeNext){
                             timeNext = localNext;
                         }
                     }
-#pragma omp barrier
+					#pragma omp barrier
                     //end Step 4
 
                 }//end simulation loop
             }
         }
 
+
+        void sequential_simulate(double timeInterval) {
```

Comment:
Same here, do we need to provide sequential simulation when the user selects a parallel coordinator?